In [1]:
import pandas as pd

In [2]:
sales = pd.read_csv("sales.csv")

In [3]:
%%sql
SELECT
    payment,
    COUNT(*) AS order_count
FROM
    sales
GROUP BY
    payment
ORDER BY
    order_count DESC;

payment  order_count
0  Credit card          659
1     Transfer          225
2         Cash          116

In [4]:
%%sql
SELECT
    payment,
    AVG(total) AS avg_order_value,
    AVG(quantity) AS avg_quantity_ordered
FROM
    sales
GROUP BY
    payment;

payment  avg_order_value  avg_quantity_ordered
0  Credit card       167.331669              5.444613
1     Transfer       709.521467             23.022222
2         Cash       165.509483              5.405172

In [5]:
%%sql
SELECT
    payment,
    AVG(total) AS avg_order_value,
    AVG(payment_fee) AS avg_payment_fee,
    SUM(total) AS total_revenue,
    SUM(payment_fee) AS total_payment_fees,
    SUM(total) - SUM(payment_fee) AS revenue_after_fees
FROM
    sales
GROUP BY
    payment;

payment  avg_order_value  avg_payment_fee  total_revenue  \
0  Credit card       167.331669             0.03      110271.57   
1     Transfer       709.521467             0.01      159642.33   
2         Cash       165.509483             0.00       19199.10   

   total_payment_fees  revenue_after_fees  
0               19.77           110251.80  
1                2.25           159640.08  
2                0.00            19199.10

# Payment Method Insights

This report provides an analysis of payment methods used for making orders and their impact on total revenue based on the provided dataset.

## Overview

The Payment Method Insights aim to understand the distribution of payment methods used by customers and analyze how payment fees impact total revenue. By examining payment methods and their associated fees, businesses can optimize payment processing strategies to minimize fees and maximize revenue.

## Findings

### Most Commonly Used Payment Methods

Based on the dataset, the most commonly used payment methods for making orders are as follows:

1. **Credit Card**: 659 orders
2. **Transfer**: 225 orders
3. **Cash**: 116 orders

These findings provide insights into the distribution of payment methods used by customers, with credit card being the most preferred method followed by transfer and cash.

### Impact of Payment Fees on Total Revenue

| Payment Method | Average Order Value ($) | Average Payment Fee ($) | Total Revenue ($) | Total Payment Fees ($) | Revenue After Deducting Payment Fees ($) |
|----------------|-------------------------|-------------------------|--------------------|------------------------|-----------------------------------------|
| Credit Card    | $167.33                 | $0.03                   | $110,271.57        | $19.77                 | $110,251.80                             |
| Transfer       | $709.52                 | $0.01                   | $159,642.33        | $2.25                  | $159,640.08                             |
| Cash           | $165.51                 | $0.00                   | $19,199.10         | $0.00                  | $19,199.10                              |

- Credit card payments have higher payment fees compared to Transfer payments, resulting in slightly lower revenue after deducting payment fees.
- Transfer payments have lower payment fees and relatively higher revenue after deducting payment fees compared to Credit card payments.
- Cash payments have no payment fees associated, resulting in the same total revenue as revenue after deducting payment fees.

### Optimization Strategies

- Encouraging the use of payment methods with lower fees, such as Transfer, can help minimize payment fees and maximize revenue.
- Negotiating lower payment processing fees or exploring alternative payment processing solutions can also reduce overall payment costs and increase revenue.
- Offering incentives or discounts for customers who use payment methods with lower fees can help drive adoption and optimize payment methods further.